src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -44,23 +45,31 @@ const WriteExerciseScreen = ({ navigation, route }: WriteExerciseScreenPropsType
   const { trainingSet, trainingSetId, disciplineTitle } = extraParams
   const [isLoading, setIsLoading] = useState(true)
   const [currentDocumentNumber, setCurrentDocumentNumber] = useState(0)
-
-  let documents = useLoadDocuments(trainingSetId).data
+  const [docs, setDocs] = useState<DocumentType[]>([])
+  const documents = useLoadDocuments(trainingSetId).data
 
   useEffect(() => {
     AsyncStorage.setSession(route.params).catch(e => console.error(e))
   }, [route.params])
 
+
+  useEffect(()=>{
+    if (documents && documents.length !== 0 && docs.length===0){
+      setDocs(documents)
+    }
+  }, [documents])
+
```

Comment:
I think you should change this to:
`  useEffect(()=>{
    if (documents.data){
      setDocs(documents.data)
    }
  }, [documents.loading])`

Makes more sense and also avoids the rerender-loop.

Reply:
Thanks that improves readability a lot